examples/ex71.cpp

```diff
@@ -0,0 +1,385 @@
+//           Hessian) are evaluated using AD.
+//
+//           We recommend viewing examples 1 and 19, before viewing this
+//           example.
+
+#include "ex71.hpp"
+
+using namespace mfem;
+
+#undef MFEM_USE_SUITESPARSE
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options
+   const char *mesh_file = "../data/beam-tet.mesh";
+   int ser_ref_levels = 3;
+   int order = 1;
+   bool visualization = true;
+   double newton_rel_tol = 1e-4;
+   double newton_abs_tol = 1e-6;
+   int newton_iter = 500;
+   int print_level = 0;
+   double pp = 2.0;
+   int integrator = 1; // use AD
+   StopWatch *timer = new StopWatch();
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh", "Mesh file to use.");
+   args.AddOption(&ser_ref_levels,
+                  "-rs",
+                  "--refine-serial",
+                  "Number of times to refine the mesh uniformly in serial.");
+   args.AddOption(&order,
+                  "-o",
+                  "--order",
+                  "Order (degree) of the finite elements.");
+   args.AddOption(&visualization,
+                  "-vis",
+                  "--visualization",
+                  "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&newton_rel_tol,
+                  "-rel",
+                  "--relative-tolerance",
+                  "Relative tolerance for the Newton solve.");
+   args.AddOption(&newton_abs_tol,
+                  "-abs",
+                  "--absolute-tolerance",
+                  "Absolute tolerance for the Newton solve.");
+   args.AddOption(&newton_iter,
+                  "-it",
+                  "--newton-iterations",
+                  "Maximum iterations for the Newton solve.");
+   args.AddOption(&pp,
+                  "-pp",
+                  "--power-parameter",
+                  "Power parameter (>=2.0) for the p-Laplacian.");
+   args.AddOption((&print_level), "-prt", "--print-level", "Print level.");
+   args.AddOption(&integrator,
+                  "-int",
+                  "--integrator",
+                  "Integrator 0: standard; 1: AD;");
+
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(std::cout);
+      return 1;
+   }
+   args.PrintOptions(std::cout);
+
+   // 2. Read the (serial) mesh from the given mesh file.
+   Mesh *mesh = new Mesh(mesh_file, 1, 1);
+   int dim = mesh->Dimension();
+
+   // 3. Refine the mesh in serial to increase the resolution. In this example
+   //    we do 'ser_ref_levels' of uniform refinement, where 'ser_ref_levels' is
+   //    a command-line parameter.
+   for (int lev = 0; lev < ser_ref_levels; lev++)
+   {
+      mesh->UniformRefinement();
+   }
+
+   // 4. Define the power parameter for the p-Laplacian and all other
+   //    coefficients
+   ConstantCoefficient c_pp(pp);
+   ConstantCoefficient load(1.000000000);
+   ConstantCoefficient c_ee(0.000000001);
```

Comment:
Are `1.0` and `1e-9` sufficient?

Reply:
They are just parameters - they will define the equation which is solved.